### PR TITLE
feat: mf-4178 new  data recovery

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-4178

### CODE REVIEW

File: packages/dashboard/src/components/Restore/CodeValidation.tsx [hunk 1]
```diff
@@ -1,47 +1,122 @@
...
-            </Step>
-        </Stepper>
+        <Box>
+            <RadioGroup
+                className={classes.purposes}
+                value={purpose}
+                onChange={(e) => {
+                    const accountType = e.currentTarget.value as AccountType
+                    setStep({ name: StepMap[accountType] })
+                    return setPurpose(accountType)
```
Comment (guanbinrui) [inline packages/dashboard/src/components/Restore/CodeValidation.tsx]:
```suggestion
                    setPurpose(accountType)
```

File: packages/dashboard/src/components/Restore/CodeValidation.tsx [hunk 2]
```diff
@@ -1,47 +1,122 @@
...
+    const { classes, theme } = useStyles()
     const [{ loading: fetchingBackupInfo }, fetchDownloadLinkFn] = useAsyncFn(
-        async (account: string, type: AccountType, code: string) => {
-            return fetchDownloadLink({ code, account, type })
-        },
+        (account: string, type: AccountType, code: string) => fetchDownloadLink({ code, account, type }),
         [],
     )
+    const [step, setStep] = useState<StepperProps['step']>({ name: ValidationCodeStep.EmailInput })
+    const [purpose, setPurpose] = useState<AccountType>(AccountType.Email)
```
Comment (guanbinrui) [inline packages/dashboard/src/components/Restore/CodeValidation.tsx]:
```suggestion
    const [purpose, setPurpose] = useState(AccountType.Email)
```

File: packages/dashboard/src/components/Restore/RestoreFromLocal.tsx [hunk 3]
```diff
@@ -77,79 +96,101 @@ export const RestoreFromLocal = memo(() => {
         if (!file) return
 
         try {
-            const decrypted = await decryptBackup(encode(password), await file.arrayBuffer())
-            setBackupValue(JSON.stringify(decode(decrypted)))
+            setReadingFile(true)
+            const [decrypted] = await Promise.all([
```
Comment (guanbinrui) [inline packages/dashboard/src/components/Restore/RestoreFromLocal.tsx:100]: Use `timeout` from `'@masknet/kit'`.

File: packages/shared/src/UI/components/UploadDropArea/index.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        const selectFile = useCallback(() => {
+            const input = document.createElement('input')
+            input.type = 'file'
+            input.hidden = true
+            input.addEventListener('input', function onInput(event) {
+                handleFiles((event.currentTarget as any).files as FileList)
+                input.removeEventListener('input', onInput)
+            })
+            input.click()
+            document.body.append(input)
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/UploadDropArea/index.tsx:97]: Need to remove the element from the body node.

File: packages/dashboard/src/components/Restore/RestoreFromLocal.tsx (hunk 3 above)
Comment (UncleBill) [reply packages/dashboard/src/components/Restore/RestoreFromLocal.tsx:100]: No, it's not for timeout, but at least waiting time for fake processing.

**Review by UncleBill — COMMENTED** [review]
PTAL @guanbinrui

File: patches/guess-calling-code@1.1.7.patch [hunk 5]
```diff
@@ -0,0 +1,24 @@
+diff --git a/package.json b/package.json
+index b98cf14b95fc3b9c8ba28ba7c2a1ca63a0140b72..b548f4a50fd33ddedf5b4ddb774966df658bd8de 100644
+--- a/package.json
++++ b/package.json
+@@ -20,6 +20,19 @@
+   "module": "dist/index.esm.js",
+   "browser": "dist/index.umd.js",
+   "types": "dist/index.d.ts",
++  "type": "module",
```
Comment (Jack-Works) [inline patches/guess-calling-code@1.1.7.patch:9]: please open an issue in that repo and add the link to ./readme.md

**Review by Jack-Works — CHANGES_REQUESTED** [review]

**Review by guanbinrui — DISMISSED** [review]

**Review by guanbinrui — APPROVED** [review]

File: packages/dashboard/src/components/Restore/steps/ValidationAccount.tsx [hunk 6]
```diff
@@ -25,6 +25,9 @@ interface ValidationAccountProps extends StepCommonProps {
     >
 }
 
+/**
+ * @deprecated Unused
```
Comment (Jack-Works) [inline packages/dashboard/src/components/Restore/steps/ValidationAccount.tsx:29]: then why not remove it?

**Pull request merged**